src/librustc/hir/lowering/expr.rs

```diff
@@ -235,11 +235,20 @@ impl LoweringContext<'_> {
     /// ```
     fn lower_expr_let(&mut self, span: Span, pat: &Pat, scrutinee: &Expr) -> hir::ExprKind {
         // If we got here, the `let` expression is not allowed.
-        self.sess
-            .struct_span_err(span, "`let` expressions are not supported here")
-            .note("only supported directly in conditions of `if`- and `while`-expressions")
-            .note("as well as when nested within `&&` and parenthesis in those conditions")
-            .emit();
+
+        if self.sess.parse_sess.unstable_features.is_nightly_build() {
```

Comment:
```suggestion
        if self.sess.opts.unstable_features.is_nightly_build() {
```
(Same thing but already public so you don't need to change the other one to `pub`... not sure why we have more than 1 of these tho; cc @estebank)

Reply:
Thanks; Squashing the commits into one seems like a good idea.